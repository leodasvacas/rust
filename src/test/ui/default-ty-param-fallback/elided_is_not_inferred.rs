// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.
// compile-flags: --error-format=human

#![feature(default_type_parameter_fallback)]

struct Bar<T>(T);

impl Bar<T> {
    fn new<U, Z:Default=String>() -> Bar<Z> {
        Bar(Z::default())
    }
}

fn main() {
    let _:u32 = foo::<usize>();
    let _:Bar<u32> = Bar::new::<usize>();
}

fn foo<U:Default, T:Default=String>() -> T {
    T::default()
}

