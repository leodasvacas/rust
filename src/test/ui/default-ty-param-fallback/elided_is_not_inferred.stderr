error[E0412]: cannot find type `T` in this scope
  --> $DIR/elided_is_not_inferred.rs:16:10
   |
16 | impl Bar<T> {
   |          ^ not found in this scope

error[E0308]: mismatched types
  --> $DIR/elided_is_not_inferred.rs:23:17
   |
23 |     let _:u32 = foo::<usize>();
   |                 ^^^^^^^^^^^^^^ expected u32, found struct `std::string::String`
   |
   = note: expected type `u32`
              found type `std::string::String`

error[E0308]: mismatched types
  --> $DIR/elided_is_not_inferred.rs:24:22
   |
24 |     let _:Bar<u32> = Bar::new::<usize>();
   |                      ^^^^^^^^^^^^^^^^^^^ expected u32, found struct `std::string::String`
   |
   = note: expected type `Bar<u32>`
              found type `Bar<std::string::String>`

error: aborting due to 3 previous errors

